### DISCUSSION TITLE
How to create a liquidity pool in uniswap v3 with the mock tokens

### DISCUSSION THREAD

**Discussion by 0xkumar** [Cyfrin/foundry-full-course-cu#1471] [Q&A]
Hello guys, I just want to create a pool with some mock tokens ,I have tried in the google and youtube for the references to create a token pool ,but all the references are using hardhat ,but im not familiar with the hardhat.Can anyone tell me how to create a pool with mock tokens using foundry

**Comment by EngrPips** [comment]
Hello @0xkumar , I think the `uniswapV3`  book would be a great resources for you as a minimized version of the whole`uniswapV3` was built in the book and the framework used is foundry 

Here is the link https://uniswapv3book.com/ to the resources. Enjoy 😉

**Comment by Vitarefiev** [comment]
Can you tell me which book to read on Defi?

**Reply by 0xkumar** [reply]
https://github.com/Jeiwan/uniswapv3-book

**Comment by Berimor11** [comment]
Good question

**Comment by yoramp44** [comment]
Thanks so much for the tip!
